features/48-user-profile-discovery/field-based-acl.md

```diff
@@ -107,10 +107,50 @@ function recordOperationVeridation (operation) {
 }
 ```
 
+## Partial Record Fields Update
+
+For backward compatibility of Skygear API, partial record fields update is
+introduced. For non-atomic record save request, partial fields update is
+allowed, which means that even some record fields are not writable for the
+requesting user, the record will still be updated on other writable fields.
+An warning message would be responded to developers showing which field updates
+are rejected because of the field-based ACL.
+
 # Changes on SDK
 
-No changes on SDK since this feature would be expected to be used on Skygear
-Cloud Portal or Skygear CLI, which will be specified on another document.
+Skygear SDKs have not many changes since the setup of field-based ACL would be
+expected on Skygear Cloud Portal or Skygear CLI, which will be specified on
+another document.
+
+The record save API on Skygear SDKs would be updated to support partial record
+fields update on non-atomic request.
+
+```js
+// saving one record
+skygear.publicDB.query(record1)
+.then((savedRecord) => {
```

Comment:
I also think that is a better interface. However, limited by the interface of [`Promise.prototype.then()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/then), the resolve function can only take one argument. We cannot add the rejected fields as an error on the 2nd argument unless we override the `Promise.prototype.then()`.